src/pages/index.js

```diff
@@ -48,20 +51,72 @@ const cardData = {
 };
 
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", (name, link) => {
-    viewImagePopup.open({ name, link });
+  // const card = new Card(cardData, "#card-template", (name, link) => {
+  //   viewImagePopup.open({ name, link });
+  // });
+  const card = new Card({
+    cardData,
+    cardSelector: "#card-template",
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleDeleteClick,
   });
+
   return card.getView();
 }
 
-const section = new Section({
-  items: initialCards,
-  renderer: (cardData) => {
-    const card = renderCard(cardData);
-    section.addItem(card);
-  },
-});
-section.renderItems();
+function handleCardClick(name, link) {
+  viewImagePopup.open({ name, link });
+}
+/* ------------------------ delete card confirmation ------------------------ */
+
+const deleteCardConfirmation = new PopupWithConfirmation("#delete-card-popup");
+
+function handleDeleteClick(card) {
+  deleteCardConfirmation.open();
+  deleteCardConfirmation.setSubmitAction(() => {
+    const id = card.getId();
+    deleteCardConfirmation.renderLoading(true, "Saving...");
+    api
+      .deleteCard(id)
+      .then(() => {
+        // deleteCardConfirmation.renderLoading(true, "Saving...");
+        deleteCardConfirmation.close();
+        card.deleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        deleteCardConfirmation.renderLoading(false);
+      });
+  });
+}
+deleteCardConfirmation.setEventListeners();
+/* ------------------------------- like button ------------------------------ */
+
+function handleLikeClick(card) {
+  if (card.isLiked()) {
+    api
+      .deleteLike(card.getID())
+      .then(() => {
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .addLike(card.getId())
+      .then(() => {
+        card.handleLikeIcon();
```

Comment:
Needs correction: there is an error in the console when I click on the like button `[webpack-dev-server] Live Reloading enabled.
2index.js:116 TypeError: card.handleLikeIcon is not a function
    at index.js:113:14`